### ISSUE TITLE
planarity as dependency

### ISSUE THREAD

**Issue by tomvanmele** [compas-dev/compas#229]
`planarity` is `pip`-installable. is there a reason why we are not including it as a dependency?

**Comment by gonzalocasas** [comment]
In principle, we could add it, but it has a dependency on `cython` which is not defined in the `pip` package, so it needs to be installed separately. This is what happens when trying to `pip install` it on a clean setup:

```
Collecting planarity
  Downloading https://files.pythonhosted.org/packages/91/d6/33cc8784b375a45505326e28cc01ed987e29a8091aa7c70de0c10aee7704/planarity-0.4.1.tar.gz (164kB)
    100% |████████████████████████████████| 174kB 1.4MB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\gcasas\AppData\Local\Temp\pip-install-dksekeaj\planarity\setup.py", line 8, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in C:\Users\gcasas\AppData\Local\Temp\pip-install-dksekeaj\planarity\ 
```

After manually triggering the pip install of `cython` it seems to work. We can definitely add to base requirements, but when we do, let's add it to the conda recipe as well, so that the installation cython dependency is handled nicely by conda.

**Comment by tomvanmele** [comment]
so do i add that in the compas-feedstock repo > recipe > meta.yaml?

**Comment by tomvanmele** [comment]
in my fork, and then issue a pr?

**Comment by brgcode** [comment]
that == then

**Comment by gonzalocasas** [comment]
yup ;)

**Comment by tomvanmele** [comment]
i noticed the version number in the recipe is still `0.3.5`. does this have to be changed manually?
also, the checklist says to bump the build number if the version remains unchanged. the build number is `1000`. technically the version number would remain unchanged, but should i not change it to `0.3.6`? and what do i do with the build number then?

**Comment by tomvanmele** [comment]
should hte hash_value aslo be changed manually? copied from pypi?

**Comment by gonzalocasas** [comment]
> version number is the recipe is still `0.3.5`

I think you have an old clone, here it's 0.3.6: https://github.com/conda-forge/compas-feedstock
(and version and hash are updated automatically by the retrobot service)

**Comment by tomvanmele** [comment]
okay, so it seems i did not do this correctly.
on travis the build passes again.

i added
```bash
pip install Cython --install-option="--no-cython-compile"
```
for `Cython` as per the recommendation on their website.

not sure how to proceed with the `conda` thing...

**Comment by tomvanmele** [comment]
`planarity` is now a requirement on not windows

**Comment by htlcnn** [comment]
@brgcode I could not install planarity on Windows. How could I install COMPAS using `python -m pip install COMPAS` and not install `planarity`?
Here's pip output when installing planarity:
```
PS C:\Users\HTL> python -m pip install planarity
Collecting planarity
  Using cached https://files.pythonhosted.org/packages/91/d6/33cc8784b375a45505326e28cc01ed987e29a8091aa7c70de0c10aee770
4/planarity-0.4.1.tar.gz
Requirement already satisfied: setuptools in c:\program files\python3\lib\site-packages (from planarity) (40.6.2)
Requirement already satisfied: cython in c:\program files\python3\lib\site-packages (from planarity) (0.29.16)
Installing collected packages: planarity
  Running setup.py install for planarity ... error
    Complete output from command "C:\Program Files\Python3\python.exe" -u -c "import setuptools, tokenize;__file__='C:\\
Users\\HTL\\AppData\\Local\\Temp\\pip-install-prvqc0rs\\planarity\\setup.py';f=getattr(tokenize, 'open', open)(__file__)
;code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\HTL\AppD
ata\Local\Temp\pip-record-3qbx1mkx\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.6
    creating build\lib.win-amd64-3.6\planarity
    copying planarity\planarity_functions.py -> build\lib.win-amd64-3.6\planarity
    copying planarity\planarity_networkx.py -> build\lib.win-amd64-3.6\planarity
    copying planarity\__init__.py -> build\lib.win-amd64-3.6\planarity
    creating build\lib.win-amd64-3.6\planarity\tests
    copying planarity\tests\test.py -> build\lib.win-amd64-3.6\planarity\tests
    copying planarity\tests\test_planarity.py -> build\lib.win-amd64-3.6\planarity\tests
    copying planarity\tests\test_planarity_networkx.py -> build\lib.win-amd64-3.6\planarity\tests
    copying planarity\tests\__init__.py -> build\lib.win-amd64-3.6\planarity\tests
    running build_ext
    cythoning planarity/planarity.pyx to planarity\planarity.c
    C:\Program Files\Python3\lib\site-packages\Cython\Compiler\Main.py:369: FutureWarning: Cython directive 'language_le
vel' not set, using 2 for now (Py2). This will change in a later release! File: C:\Users\HTL\AppData\Local\Temp\pip-inst
all-prvqc0rs\planarity\planarity\planarity.pyx
      tree = Parsing.p_module(s, pxd, full_module_name)
    building 'planarity.planarity' extension
    error: Unable to find a compatible Visual Studio installation.

    ----------------------------------------
Command ""C:\Program Files\Python3\python.exe" -u -c "import setuptools, tokenize;__file__='C:\\Users\\HTL\\AppData\\Loc
al\\Temp\\pip-install-prvqc0rs\\planarity\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('
\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\HTL\AppData\Local\Temp\pip-recor
d-3qbx1mkx\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\HTL\Ap
pData\Local\Temp\pip-install-prvqc0rs\planarity\
```

**Comment by tomvanmele** [comment]
you could clone the source, remove planarity from the requirements and install from source.

but it seems there is a problem with your Cython installation and/or the available C/C++ compilers on your system. have you looked at the [Known Issues](https://compas-dev.github.io/main/gettingstarted/knownissues.html)?

i know you don't literally get the same errors as listed there, but the problem sounds remarkably similar. perhaps trying those fixes is also worth a shot.

**Comment by tomvanmele** [comment]
btw, any reason you are using a standard Python install (`C:\Program Files\Python3\python.exe`) and not something like `conda` or `miniconda`?

**Comment by htlcnn** [comment]
I cloned the source, removed planarity from requirements.txt. Installation was successful. Does compas run normally without planarity?
I read Known Issues and installed both Cython and VS 2019 C++ Build Tools, but somehow Cython could not find Visual Studio as it said.
I've been used to pure python installation from python.org and install packages via pip. conda seems to have many existing packages that I don't need. I tried miniconda but it could not connect to server to download packages, so I switched back to pure python and pip.
Thanks for your help!
